tests/sles4sap/monitoring_services.pm

```diff
@@ -0,0 +1,174 @@
+use hacluster qw(add_file_in_csync get_cluster_name get_hostname is_node wait_until_resources_started);
+
+sub run {
+    my ($self)        = @_;
+    my $hostname      = get_hostname;
+    my $cluster_name  = get_cluster_name;
+    my $instance_sid  = get_required_var('INSTANCE_SID');
+    my $instance_type = get_required_var('INSTANCE_TYPE');
+    my $instance_id   = get_required_var('INSTANCE_ID');
+    my $rsc_id        = "${instance_sid}_${instance_type}${instance_id}";
+
+    $self->select_serial_terminal;
+
+    # Configure HA Exporter
+    if (get_var('HA_CLUSTER')) {
+        my $ha_exporter_config = 'ha_cluster_exporter';
+
+        # Install needed packages
+        zypper_call "in prometheus-$ha_exporter_config";
+
+        # Get the IP port and start exporter
+        my ($ha_exporter_port) = script_output("awk '/port:/ { print \$NF }' /etc/$ha_exporter_config /usr/etc/$ha_exporter_config 2>/dev/null", proceed_on_failure => 1) =~ /(\d+)/;
+        $ha_exporter_port ||= 9664;
+        systemctl "enable --now prometheus-$ha_exporter_config";
+        systemctl "status prometheus-$ha_exporter_config";
+        assert_script_run "curl http://localhost:$ha_exporter_port";
+    }
+
+    # Configure HANADB Exporter
+    if (get_var('HANA')) {
+        my $hanadb_exporter_config = "/usr/etc/hanadb_exporter/$rsc_id.json";
+        my $check_exporter         = 'false';
+        my $hanadb_exporter_port;
+
+        # Install needed packages (hdcli contains dbapi)
+        zypper_call 'in prometheus-hanadb_exporter';
+        assert_script_run "pip install \$(ls /hana/shared/${instance_sid}/hdbclient/hdbcli-*.tar.gz)";
+
+        # Modify the configuration file
+        assert_script_run "cp /usr/etc/hanadb_exporter/config.json.example $hanadb_exporter_config";
+        file_content_replace("$hanadb_exporter_config",
+            q(localhost)            => $hostname,
+            q(PASSWORD)             => $sles4sap::instance_password,
+            q(./logging_config.ini) => $hanadb_exporter_config,
+            q(hanadb_exporter.log)  => "/var/log/hanadb_exporter_$rsc_id.log"
+        );
+
+        # Get the IP port and start exporter
+        ($hanadb_exporter_port) = script_output("awk '/exposition_port/ { print \$NF }' $hanadb_exporter_config", proceed_on_failure => 1) =~ /(\d+)/;
+        $hanadb_exporter_port ||= 9668;
+
+        # Add monitoring resource in the HA stack
+        if (get_var('HA_CLUSTER') and is_node(1)) {
+            my $hanadb_msl     = "msl_SAPHana_$rsc_id";
+            my $hanadb_exp_rsc = "rsc_exporter_$rsc_id";
+            $hanadb_exporter_port = 9668;
+            $check_exporter       = 'true';
+
+            # We need to add the configuration in csync2.conf
+            add_file_in_csync(value => "$hanadb_exporter_config");
+
+            # Add the monitoring resource
+            assert_script_run(
+                'crm configure primitive ' . $hanadb_exp_rsc . ' systemd:prometheus-hanadb_exporter@' . $rsc_id
```

Comment:
Good idea, we should rework all of our cluster commands like that if it works on 12-SP2.